extras/asr-services/common/base_service.py

```
@@ -97,12 +99,47 @@ def get_supported_languages(self) -> Optional[list[str]]:
     @property
     def is_ready(self) -> bool:
         """Return True if service is ready to handle requests."""
         return self._is_ready
 
 
+def _get_audio_duration(file_path: str) -> Optional[float]:
+    """Return audio duration in seconds, or None if unreadable."""
+    try:
+        with wave.open(file_path, "rb") as wf:
+            return wf.getnframes() / wf.getframerate()
+    except Exception:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)